### DISCUSSION TITLE
Package com.cburch.gray never referenced?

### DISCUSSION THREAD

**Discussion by zdimension** [logisim-evolution/logisim-evolution#600] [Dev talk]
I stumbled upon that package while browsing the code, it seems to contains three components related to Gray codes, but the classes are never referenced in the project so the components are not available. Is there something I'm missing?

I think the com.cburch.gray package would fit better under com.cburch.logisim.std.*

**Comment by BFH-ktt1** [comment]
This package is an example project for the "jar-library" import function. It is for the moment still outdated and should be updated. You can find information on this package in the help under "guide to Being a Logisim User -> JAR libraries".
I agree that the place is not the best one, however, this is still a part of the code that needs to be addressed and checked.
I'm open for sugestions

**Comment by R3dst0ne** [comment]
Since Logisim-evolution now has its own organization. What is your opinion on moving the Gray library to its own repository?
This would help people who want to develop their own librarys, since they then simply can clone it and are ready to start/modify.

**Reply by maehne** [reply]
This could be an option -- especially if we can also provide a Gradle-based build system as part of the example. Good up-to-date documentation at least in English will be the key issue that it's going to be useful to someone. Maybe, the new repository could become an (optional?) sub-module of the current Git repository that would then get automatically built and integrated into Logisim-evolution. It could maybe show 3rd-party developers how to contribute their own libraries in the mid-term to long-term. Before going in this direction. Let's hear what the others are thinking. For me, it would not be of highest priority. One thing to consider is also that some people struggle with Git. Adding sub-modules would complicate things further to them.

**Reply by BFH-ktt1** [reply]
I'm in favor. As this part is an example how to make a jar library that can be included in logisim. However, I have no slightest idea what the status is of the jar-library source code, as it hasn't be touched since the original version of Carl Burch. I can well imagine that it is outdated and should be "revised". Also this we should discuss in the roadmap discussion.

**Comment by R3dst0ne** [comment]
Can someone, with experience in `Gradle`, guide me how to set it up, so that I can work on the `Gray librarie`?

**Comment by MarcinOrlowski** [comment]
The major obstacle to fully separate `gray` from main code is that we have no SDK that we can use as bridge.

**Reply by maehne** [reply]
So, how about setting it first up as part of the current Gradle build system? Then, we could make sure that at least the Gray librarie is still compatible with Logisim-evolution.
